hal/tests/TESTS/pin_names/pinvalidate.py

```diff
@@ -14,318 +14,416 @@
+        )
+
+    return tabulate(
+        table,
+        headers=["Platform name", "Result", "Error count"],
+        tablefmt="grid",
+    )
+
 
 def print_suite_summary(report):
-    targets = []
-    for case in report:
-        targets.append(case['platform_name'])
-    targets = list(dict.fromkeys(targets))
-    
+    targets = set([case["platform_name"] for case in report])
+
     table = []
     for target in targets:
-        suites = []
-        for case in report:
-            if case['platform_name'] != target:
-                continue
-            suites.append(case['suite_name'])
-        suites = list(dict.fromkeys(suites))
+        suites = set([
+            case["suite_name"]
+            for case in report
+            if case["platform_name"] == target
+        ])
 
         for suite in suites:
-            result = 'PASSED'
+            result = "PASSED"
             error_count = 0
             for case in report:
-                if case['platform_name'] != target:
+                if case["platform_name"] != target:
                     continue
-                if case['suite_name'] != suite:
+                if case["suite_name"] != suite:
                     continue
-                if case['result'] == 'FAILED':
-                    result = 'FAILED'
+                if case["result"] == "FAILED":
+                    result = "FAILED"
                     error_count += 1
             table.append([target, suite, result, error_count])
-    
-    return tabulate(table, headers=['Platform name', 'Test suite', 'Result', 'Error count'], tablefmt='grid')
 
-def print_report(report, print_error_detail, tablefmt='grid'):
+    return tabulate(
+        table,
+        headers=["Platform name", "Test suite", "Result", "Error count"],
+        tablefmt="grid",
+    )
+
+
+def print_report(report, print_error_detail, tablefmt="grid"):
     table = []
     for case in report:
         errors_str = ""
-        for error in case['errors']:
-            errors_str += '\n'
-            if error['key']:
-                errors_str += error['key']
-            if error['val']:
-                errors_str += " = " + error['val']
-            if error['message']:
-                errors_str += " <-- " + error['message']
-        
-        if errors_str == '':
-            errors_str = 'None'
+        for error in case["errors"]:
+            errors_str += "\n"
+            if error["key"]:
+                errors_str += error["key"]
+            if error["val"]:
+                errors_str += " = " + error["val"]
+            if error["message"]:
+                errors_str += " <-- " + error["message"]
+
+        if errors_str == "":
+            errors_str = "None"
 
         if print_error_detail:
-            table.append([case['platform_name'], case['suite_name'], case['case_name'], case['result'], len(case['errors']), errors_str.lstrip()])
+            table.append(
+                [
```

Comment:
@gpsimenos Please change this to be a tuple like in the `else`.

Reply:
Done